snuba/clickhouse/formatter.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional, Sequence
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    ExpressionVisitor,
+    FunctionCall,
+    Literal,
+)
+from snuba.query.parsing import ParsingContext
+from snuba.util import escape_alias, escape_col, escape_string
+
+# Tokens used when formatting. Defining them as costant
+# will make it easy (if/when needed) to make some changes
+# to the layout of the output.
+NULL = "NULL"
+COMMA = ","
+SPACE = " "
+TRUE = "true"
+FALSE = "false"
+QUOTE = "'"
+OPEN_BRACKET = "("
+CLOSED_BRACKET = ")"
+DOT = "."
```

Comment:
I feel like this makes formatting more challenging understand at a glance, but if you think the flexibility is worth that disadvantage, feel free to leave as-is, I don't have a strong opinion one way or the other here.

Reply:
Let me try a commit in which I keep only SPACE as a token. It will be trivial. To be fair I did not put too much thinking into this, I was thinking about a reversed tokenization. Maybe it is overkill.